src/DynamoPackages/PackageDownloadHandle.cs

```diff
@@ -29,16 +29,21 @@ public State DownloadState
             }
         }
 
+        [Obsolete("No longer used. Remove in 3.0")]
         public Greg.Responses.PackageHeader Header { get; private set; }
         private string _name;
         public string Name { get { return Header != null ? Header.name : _name; } set { _name = value; } }
 
+        private string _id;
+        public string Id { get { return Header != null ? Header._id : _id; } set { _id = value; } }
```

Comment:
for public properties please add /// summaries - also, should this really be public? Do we intend for others extensions/integrators to use this?

Reply:
This is used in `DynamoCoreWpf` so I'll add comments for it. I don't think someone might want to extend it in some way. This is used as a view model for the installed dependencies when installing a package from the search.

Reply:
if it's only to be used for us - maybe we can make it internal and use the internalsVisibleTo attribute, my feeling is that we should be much more deliberate about what constitutes our public api. What do you think?

Reply:
My opinion is that using `InternalsVisibleTo` is more of a hack than a good practice. Making something public shouldn't make it "written in stone". I think we should identify exactly what we want our extension points to be instead, and only make promises for those.